src/npctalk.cpp

```diff
@@ -5585,29 +5586,63 @@ void talk_effect_fun_t::set_spawn_monster( const JsonObject &jo, std::string_vie
-                            dov_max_radius, outdoor_only, indoor_only, group, dov_lifespan, target_var,
+                            dov_max_radius, outdoor_only, indoor_only, group, single_target, dov_lifespan, target_var,
                             spawn_message, spawn_message_plural, true_eocs, false_eocs, open_air_allowed,
                 friendly, is_npc]( dialogue & d ) {
         monster target_monster;
+        std::vector<Creature *> target_monsters;
+        mongroup_id target_mongroup;
 
         if( group ) {
-            target_monster = monster( MonsterGroupManager::GetRandomMonsterFromGroup( mongroup_id(
-                                          monster_id.evaluate( d ) ) ) );
+            if( monster_id.evaluate( d ).empty() ) {
+                debugmsg( "Cannot use group without a valid monstergroup.  %s", d.get_callstack() );
+            }
+            if( single_target ) {
+                target_monster = monster( MonsterGroupManager::GetRandomMonsterFromGroup( mongroup_id(
+                                              monster_id.evaluate( d ) ) ) );
+            } else {
+                target_mongroup = mongroup_id( monster_id.evaluate( d ) );
+            }
         } else if( monster_id.evaluate( d ).empty() ) {
             int target_range = dov_target_range.evaluate( d );
-            //grab a random nearby hostile creature to create a hallucination or copy of
-            Creature *copy = g->get_creature_if( [target_range]( const Creature & critter ) -> bool {
-                bool not_self = get_player_character().pos() != critter.pos();
-                bool in_range = std::round( rl_dist_exact( get_player_character().pos(), critter.pos() ) ) <= target_range;
-                bool valid_target = get_player_character().attitude_to( critter ) == Creature::Attitude::HOSTILE;
-                return not_self && in_range && valid_target;
-            } );
-            if( copy == nullptr ) {
-                run_eoc_vector( false_eocs, d );
-                return;
+            if( single_target ) {
+                // Grab a random nearby hostile creature to create a hallucination or copy of
+                Creature *copy = g->get_creature_if( [target_range]( const Creature & critter ) -> bool {
+                    bool not_self = get_player_character().pos() != critter.pos();
+                    bool in_range = std::round( rl_dist_exact( get_player_character().pos(), critter.pos() ) ) <= target_range;
+                    bool valid_target = get_player_character().attitude_to( critter ) == Creature::Attitude::HOSTILE;
+                    return not_self && in_range && valid_target;
+                } );
+                if( copy == nullptr ) {
+                    run_eoc_vector( false_eocs, d );
+                    return;
+                }
+                target_monster = *copy->as_monster();
+            } else {
+                // Grab all hostile creatures to create hallucinations or copies of
```

Comment:
`u_spawn_monster` is not strictly about hallucinations

Reply:
Ik, that part is only if using "u/"npc_spawn_monster": "" and specifies hallucinations or copies not just hallucinations